src/waveform/renderers/waveformmark.h

```diff
@@ -22,6 +22,11 @@ class WaveformMark {
         // To indicate that the image for the mark needs to be regenerated,
         // when the text, color, breadth or level are changed.
         bool m_obsolete{};
+        Graphics() = default;
+        virtual ~Graphics() = default;
+        // non-copyable and non-movable
+        Graphics(const Graphics&) = delete;
+        Graphics& operator=(const Graphics&) = delete;
     };
```

Comment:
thats just copy, not move though...

```suggestion
        Graphics(Graphics&&) = delete;
        Graphics& operator=(Graphics&&) = delete;
    };
```